spyder/plugins/updatemanager/workers.py

```diff
@@ -220,77 +350,73 @@ def _check_update_available(
+            asset_info = get_asset_info(releases.get(latest_release))
+
+            if asset_info is not None:
+                # The asset is available, now get checksum.
+                checksum = get_asset_checksum(
+                    asset_info["checksum_url"], asset_info["filename"]
+                )
+                if checksum is not None:
+                    asset_info.update(checksum=checksum)
+                    logger.debug(f"Asset available: {latest_release}")
+                    break
                 else:
-                    # The asset is not available
-                    logger.debug(
-                        "Asset not available: "
-                        f"{page.status_code} Client Error: {page.reason} "
-                        f"for url: {page.url}"
-                    )
-                    asset_available = False
-                    releases.remove(latest_release)
+                    asset_info = None
+
+            # The asset is not available
+            logger.debug(f"Asset not available: {latest_release}")
+            releases.pop(latest_release)
 
-                    latest_release = max(releases) if releases else CURRENT_VERSION
-                    update_available = CURRENT_VERSION < latest_release
+            latest_release = max(releases) if releases else CURRENT_VERSION
+            update_available = CURRENT_VERSION < latest_release
 
-                    logger.debug(f"Latest release: {latest_release}")
-                    logger.debug(f"Update available: {update_available}")
+            logger.debug(f"Latest release: {latest_release}")
+            logger.debug(f"Update available: {update_available}")
 
-        self.latest_release = latest_release
-        self.update_available = update_available
+        self.asset_info = asset_info
 
     def start(self):
         """Main method of the worker."""
         self.error = None
-        self.latest_release = None
-        self.update_available = False
+        self.asset_info = None
         error_msg = None
-        url = 'https://api.github.com/repos/spyder-ide/spyder/releases'
 
+        url = None
         if not is_conda_based_app():
-            self.channel = "pypi"  # Default channel if not conda
             if is_conda_env(sys.prefix):
-                self.channel, channel_url = get_spyder_conda_channel()
-
-            # If Spyder is installed from defaults channel (pkgs/main), then
-            # use that channel to get updates. The defaults channel can be far
-            # behind our latest release.
-            if self.channel == "pkgs/main":
-                url = channel_url + '/channeldata.json'
-        github = "api.github.com" in url
-
-        headers = {}
-        token = os.getenv('GITHUB_TOKEN')
-        if running_in_ci() and token:
-            headers.update(Authorization=f"Bearer {token}")
+                channel, url = get_spyder_conda_channel()
+                url += '/channeldata.json'
```

Comment:
```suggestion
                # This covers the case of Spyder being installed with pip in a
                # conda environment
                if channel == "pypi":
                    url = "https://pypi.python.org/pypi/spyder/json"
                else:
                    url += '/channeldata.json'
```

This is my case and I was getting an error due to it not being handled correctly.

Reply:
Good suggestion. Permit me to reduce the nested conditionals...